src/main/java/core/basesyntax/SquareSupplier.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+public class SquareSupplier {
+    private static final RandomGeometricValue randomSize = new RandomGeometricValue();
+    private static final ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomSquare() {
+        return new Square(randomSize.getRandomGeometricValue(), color.getRandomColor());

```

Comment:
The method 'getRandomSquare' should be static. As it produces a new Square object every time it's called, it doesn't make sense to have it as an instance method.

Reply:
When I have changed it got an information about error. What should I do with that?